smart_selects/widgets.py

```diff
@@ -250,28 +258,31 @@ def render(self, name, value, attrs=None, choices=()):
             auto_choose = 'true'
         else:
             auto_choose = 'false'
-        js = """
-        <script type="text/javascript">
-        (function($) {
-
-        var chainfield = "#id_%(chainfield)s";
-        var url = "%(url)s";
-        var id = "#%(id)s";
-        var value = %(value)s;
-        var auto_choose = %(auto_choose)s;
-        // Use $(window).load to call function after SelectBox and SelectFilter2
-        $(window).load(function() {
-            chainedm2m.init(chainfield, url, id, value, auto_choose);
-        });
-        })(jQuery || django.jQuery);
-        </script>
-
-        """
-        js = js % {"chainfield": chain_field,
-                   "url": url,
-                   "id": attrs['id'],
-                   'value': '""' if value is None else json.dumps(value),
-                   'auto_choose': auto_choose}
+        # js = """
+        # <script type="text/javascript">
+        # (function($) {
+        # var chainfield = "#id_%(chainfield)s";
+        # var url = "%(url)s";
+        # var id = "#%(id)s";
+        # var value = %(value)s;
+        # var auto_choose = %(auto_choose)s;
+        # // Use $(window).load to call function after SelectBox and SelectFilter2
+        # $(window).load(function() {
+        #     chainedm2m.init(chainfield, url, id, value, auto_choose);
+        # });
+        # })(jQuery || django.jQuery);
+        # </script>
+        # """
+        # js = js % {"chainfield": chain_field,
+        #            "url": url,
+        #            "id": attrs['id'],
+        #            'value': '""' if value is None else json.dumps(value),
+        #            'auto_choose': auto_choose}
+
+        attrs["data-chainfield"] = chain_field
+        attrs["data-url"] = url
+        attrs["data-value"] = '""' if value is None else json.dumps(value)
```

Comment:
This is where the `data-value="&quot;&quot;"` is coming from. This was originally changed because it was interpreted on the client side, but I think just specifying a blank string (`''`, take out the double quotes) would fix it.